tests/test_integration.c

```diff
@@ -0,0 +1,94 @@
+#include "sentry.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static void
+print_envelope(sentry_envelope_t *envelope, void *_data)
+{
+    size_t size_out = 0;
+    char *s = sentry_envelope_serialize(envelope, &size_out);
+    printf("%s", s);
+    sentry_free(s);
+}
+
+static bool
+has_arg(int argc, char **argv, const char *arg)
+{
+    for (int i = 1; i < argc; i++) {
+        if (strcmp(argv[i], arg) == 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+int
+main(int argc, char **argv)
+{
+    sentry_options_t *options = sentry_options_new();
+
+    sentry_options_set_environment(options, "Production");
+    sentry_options_set_release(options, "integration-test-release");
+
+    if (has_arg(argc, argv, "attachment")) {
+        // we run the test directly from the build directory
+        sentry_options_add_attachment(
+            options, "CMakeCache.txt", "./CMakeCache.txt");
+    }
+
+    if (has_arg(argc, argv, "stdout")) {
+        sentry_options_set_transport(
+            options, sentry_new_function_transport(print_envelope, NULL));
+    }
+
+    sentry_init(options);
+
+    if (!has_arg(argc, argv, "no-setup")) {
+        sentry_set_transaction("test-transaction");
+        sentry_set_level(SENTRY_LEVEL_WARNING);
+        sentry_set_extra("extra stuff", sentry_value_new_string("some value"));
+        sentry_set_tag("expected-tag", "some value");
+        sentry_set_tag("not-expected-tag", "some value");
+        sentry_remove_tag("not-expected-tag");
+
+        sentry_value_t user = sentry_value_new_object();
+        sentry_value_set_by_key(user, "id", sentry_value_new_int32(42));
+        sentry_value_set_by_key(
+            user, "username", sentry_value_new_string("some_name"));
+        sentry_set_user(user);
+
+        sentry_value_t default_crumb
+            = sentry_value_new_breadcrumb(0, "default level is info");
+        sentry_add_breadcrumb(default_crumb);
+
+        sentry_value_t debug_crumb
+            = sentry_value_new_breadcrumb("http", "debug crumb");
+        sentry_value_set_by_key(
+            debug_crumb, "category", sentry_value_new_string("example!"));
+        sentry_value_set_by_key(
+            debug_crumb, "level", sentry_value_new_string("debug"));
+        sentry_add_breadcrumb(debug_crumb);
+    }
+
+    if (has_arg(argc, argv, "overflow-breadcrumbs")) {
+        for (size_t i = 0; i < 101; i++) {
+            char buffer[4];
+            sprintf(buffer, "%zu", i);
+            sentry_add_breadcrumb(sentry_value_new_breadcrumb(0, buffer));
+        }
+    }
+
+    if (has_arg(argc, argv, "crash")) {
+        memset((char *)0x0, 1, 100);
```

Comment:
Having this in the example would be really great, too.

Reply:
I actually plan to remove the example completely, and replace it with the integration test file ;-)